### PR TITLE
Add hosting config page

### PR DESCRIPTION
**Pull request by eoigal**
This PR moves the hosting config from the hosting wp-admin menu to the Global site view (GSV).

Ref: https://github.com/Automattic/dotcom-forge/issues/6255

### Nav Uni - Hosting config (where it is now)

Screenshot 2024-04-22 at 11 54 33

### GSV - Hosting Config

Screenshot 2024-04-22 at 11 52 59

This page will eventually be a tab in the sites dashboard.

### Testing Instructions

Confirm hosting config loads from a URL similar to `/hosting-config/{atomic-site}`
Confirm doesn't affect the current sidebars when feature flag is disabled
To enable feature flag include `?flags=layout/dotcom-nav-redesign-v2`
To disable feature flag include `?flags=-layout/dotcom-nav-redesign-v2`

## TODO

- [x] Add handler for hosting config pages
- [x] Merge https://github.com/Automattic/wp-calypso/pull/89771 and rebase this PR
- [x] Update `getShouldShowGlobalSidebar` to support showing global sidebar when feature flag is enabled
- [ ] Possibly refactor `client/hosting-overview` to be some something like `client/hosting-tools` or just `client/hosting`

### CODE REVIEW

File: client/sections.js [hunk 1]
```diff
@@ -218,7 +218,7 @@ const sections = [
 	{
 		name: 'hosting-overview',
 		paths: [ '/hosting-overview' ],
-		module: 'calypso/hosting-overview',
+		module: 'calypso/hosting',
```
Comment (roo2) [inline client/sections.js]: I got an error when accessing `/hosting-overview` or `/hosting-config`

```
cannot find the chunk "hosting"
```

**Review by roo2 — COMMENTED** [review]
I was just looking at 
https://github.com/Automattic/wp-calypso/blob/9a4b1b9daed06680f028e61ee5d6428121c52df1/client/a8c-for-agencies/sections/sites/features/a4a/overview-preview-pane.tsx#L61

It looks like to use that a4a component we need to pass in components rather than routes. I'm not sure if we would need a new route or just a component that can be passed in to that `features` array

File: client/state/global-sidebar/selectors.ts [hunk 2]
```diff
@@ -14,14 +14,24 @@ export const getShouldShowGlobalSidebar = (
...
+	}
+	// Show the global sidebar for the sites dashboard
+	// If on hosting section, also check if the feature flag is enabled
+	if (
+		sectionGroup === 'sites-dashboard' &&
+		( sectionName !== 'hosting' || isEnabled( 'layout/dotcom-nav-redesign-v2' ) )
+	) {
+		return true;
+	}
+	return false;
```
Comment (okmttdhr) [inline client/state/global-sidebar/selectors.ts]: I understand the behavior for `sectionGroup === 'sites-dashboard'` as follows:

- When no site is selected (`/sites`), we display the global sidebar in its expanded form.
- When a site is selected, on any page being viewed (e.g., the hosting configuration page or the GitHub deployment page), we display the global sidebar in its collapsed form.

If so, could we update this entire condition in `getShouldShowGlobalSidebar` to the following? (sorry if I'm wrong)

```ts
	return (
		sectionGroup === 'me' ||
		sectionGroup === 'reader' ||
		( isEnabled( 'layout/dotcom-nav-redesign-v2' )
			? sectionGroup === 'sites-dashboard' && ! siteId
			: sectionGroup === 'sites-dashboard' ) || 
		( sectionGroup === 'sites' && ! siteId )
	);
```

The collapsed global sidebar would be displayed appropriately, as covered in `getShouldShowCollapsedGlobalSidebar` https://github.com/Automattic/wp-calypso/blob/add/hosting-config/client/state/global-sidebar/selectors.ts#L37-L47

File: client/sections.js (hunk 1 above)
Comment (eoigal) [reply client/sections.js]: Oops forgot to revert this part too. Thanks.

File: client/state/global-sidebar/selectors.ts [hunk 3]
```diff
@@ -20,7 +23,11 @@ export const getShouldShowGlobalSidebar = (
 		sectionGroup === 'me' ||
 		sectionGroup === 'reader' ||
 		sectionGroup === 'sites-dashboard' ||
-		( sectionGroup === 'sites' && ! siteId )
+		( sectionGroup === 'sites' &&
+			( ! siteId ||
+				( !! siteId &&
+					isEnabled( 'layout/dotcom-nav-redesign-v2' ) &&
+					GLOBAL_SIDEBAR_SECTION_NAMES.includes( sectionName ) ) ) )
```
Comment (okmttdhr) [inline client/state/global-sidebar/selectors.ts]: badge

For the hosting configuration page in this PR, we will eventually need to display the collapsed global sidebar along with the list of sites, as illustrated in the design below (So we'll need to tweak the condition (e.g., `getShouldShowCollapsedGlobalSidebar`) later). Is my understanding correct?
Screenshot 2024-04-24 at 11 09 48

File: client/hosting-overview/index.tsx [hunk 4]
```diff
@@ -1,9 +1,35 @@
...
+import { navigation, siteSelection, sites } from 'calypso/my-sites/controller';
+import { redirectHomeIfIneligible } from 'calypso/my-sites/github-deployments/controller';
+import { handleHostingPanelRedirect } from 'calypso/my-sites/hosting/controller';
+import { hostingOverview, hostingConfiguration, hostingActivate } from './controller';
 
 export default function () {
 	page( '/hosting-overview', siteSelection, sites, makeLayout, clientRender );
 	page( '/hosting-overview/:site', siteSelection, hostingOverview, makeLayout, clientRender );
+
+	page( '/hosting-config', siteSelection, sites, makeLayout, clientRender );
```
Comment (roo2) [inline client/hosting-overview/index.tsx:12]: hmm should these also be removed from `client/my-sites/hosting/index.js`? they seem to be duplicated now

**Review by okmttdhr — APPROVED** [review]
It worked as described ✅  I left a question about future changes.
Screenshot 2024-04-24 at 10 57 18
Screenshot 2024-04-24 at 10 57 44

File: client/state/global-sidebar/selectors.ts (hunk 2 above)
Comment (okmttdhr) [reply client/state/global-sidebar/selectors.ts]: p1713866950557289-slack-C06TF5T7MV5

File: client/state/global-sidebar/selectors.ts [hunk 5]
```diff
@@ -20,7 +23,11 @@ export const getShouldShowGlobalSidebar = (
 		sectionGroup === 'me' ||
```
Comment (roo2) [inline client/state/global-sidebar/selectors.ts:30]: nit: you can take the eslint comment off line 20 now 😄

**Review by roo2 — APPROVED** [review]
works as described and doesn't seem to affect current behaviour when the flag is disabled

File: client/hosting-overview/index.tsx (hunk 4 above)
Comment (fushar) [reply client/hosting-overview/index.tsx:12]: No, they are still required to show the pages in Nav Unification (when the flag is not enabled).

**Comment by fushar** [comment]
Pushed bcddffbaf23432a75f201bd2fafcb2e124ae5d6b to simplify the conditions, could we re-test with and without the flag? 🙏

File: client/state/global-sidebar/selectors.ts (hunk 5 above)
Comment (fushar) [reply client/state/global-sidebar/selectors.ts:30]: Done!

**Review by fushar — APPROVED** [review]

**Review by okmttdhr — APPROVED** [review]
Much easier to read, thanks @fushar!

Worked as expected, while I have a question: https://github.com/Automattic/wp-calypso/pull/89735#discussion_r1577147135

**Comment by autumnfjeld** [comment]
@fushar  My tests:

✅  Confirm hosting config loads from a URL similar to /hosting-config/{atomic-site}
with ?flags=layout/dotcom-nav-redesign-v2

image


✅  Confirm doesn't affect the current sidebars when feature flag is disabled
No feature flag
image

File: client/state/global-sidebar/selectors.ts (hunk 3 above)
Comment (miksansegundo) [reply client/state/global-sidebar/selectors.ts]: The collapsed global sidebar is shown when using the feature flag in https://github.com/Automattic/wp-calypso/pull/89741 but not yet in https://github.com/Automattic/wp-calypso/pull/89749.

I'm assuming that change is not a blocker for this PR and https://github.com/Automattic/wp-calypso/pull/89749, and will be done in a follow-up PR. cc: @eoigal  @fushar

Comment (fushar) [reply client/state/global-sidebar/selectors.ts]: Ohhh I understand now. Yeah let's do it in a follow-up PR! (I can try to take care of it). Thanks!

**Comment by okmttdhr** [comment]
Let's address non-blockers in follow-up PRs!

**Comment by miksansegundo** [comment]
Noting a possible improvement of the GitHub Deployments banner. My atomic site has a repo connected, and I have done deployments but the banner still says "Get Started". 

Screenshot 2567-04-24 at 14 13 42

Not sure if we can add a check [here](https://github.com/Automattic/wp-calypso/blob/221022601ce19963cd7ec37e4a827633005a2a02/client/my-sites/hosting/main.js#L98) to only add the card if the site doesn't have a repo connected.

I'm creating an issue in [Untangling & Nav Redesign](https://github.com/orgs/Automattic/projects/927/views/2) project. Please, let me know if I should create it somewhere else.

File: client/hosting-overview/index.tsx (hunk 4 above)
Comment (fushar) [reply client/hosting-overview/index.tsx:12]: Sorry I think you're right; they seem to be unused now. We can clean up in a follow-up PR.

**Pull request merged**